crates/ra_hir_def/src/nameres/path_resolution.rs

```diff
@@ -68,8 +68,10 @@ impl CrateDefMap {
         mode: ResolveMode,
         original_module: LocalModuleId,
         path: &Path,
+        shadow: BuiltinShadowMode,
     ) -> ResolvePathResult {
-        let mut segments = path.segments.iter().enumerate();
+        let mut segments = path.segments.iter().enumerate().peekable();
```

Comment:
the use of peekable seems a bit overkill, since you could just compare the index with the length?

Reply:
Good call ! I totally forget we have `enumerate` here. :)